src/components/button/button.spec.js

```diff
@@ -0,0 +1,364 @@
+      size: 'large',
+      children: 'Large'
+    });
+
+    disabled = render({
+      name: 'Disabled Button',
+      disabled: true,
+      children: 'Disabled'
+    });
+
+    anchor = render({
+      href: '/foo',
+      children: 'Anchor'
+    }, mount);
+
+    to = render({
+      to: '/foo',
+      children: 'To'
+    }, mount);
+  });
+
+  describe('A basic button', () => {
+    it('renders a button with defaults', () => {
+      expect(defaultButton.props().renderAs).toEqual('secondary');
+      expect(defaultButton.containsMatchingElement(
+        <span>Save</span>
+      )).toBeTruthy();
+      expect(defaultButton.props().disabled).toEqual(false);
+      expect(defaultButton.props().size).toEqual('medium');
+      expect(defaultButton.props().variant).toEqual('blue');
+    });
+  });
+
+  describe('A primary button', () => {
+    it('renders a primary button', () => {
+      expect(primary.props().name).toEqual('Primary Button');
+      expect(primary.props().renderAs).toEqual('primary');
+      expect(primary.containsMatchingElement(
+        <span>Primary</span>
+      )).toBeTruthy();
+      expect(primary.props().disabled).toEqual(false);
+    });
+  });
+
+  describe('A secondary button', () => {
+    it('renders a secondary button', () => {
+      expect(secondary.props().name).toEqual('Secondary Button');
+      expect(secondary.props().renderAs).toEqual('secondary');
+      expect(secondary.containsMatchingElement(
+        <span>Secondary</span>
+      )).toBeTruthy();
+      expect(secondary.props().disabled).toEqual(false);
+    });
+  });
+
+  describe('A small button', () => {
+    it('renders a small button', () => {
+      expect(small.props().name).toEqual('Small Button');
+      expect(small.props().size).toEqual('small');
+      expect(small.containsMatchingElement(
+        <span>Small</span>
+      )).toBeTruthy();
+      expect(small.props().disabled).toEqual(false);
+    });
+  });
+
+  describe('A large button', () => {
+    it('renders a large button', () => {
+      expect(large.props().name).toEqual('Large Button');
+      expect(large.props().size).toEqual('large');
+      expect(large.containsMatchingElement(
+        <span>Large</span>
+      )).toBeTruthy();
+      expect(large.props().disabled).toEqual(false);
+    });
+  });
+
+  describe('A disabled button', () => {
+    it('renders a disabled button', () => {
+      expect(disabled.props().name).toEqual('Disabled Button');
+      expect(disabled.props().renderAs).toEqual('secondary');
+      expect(disabled.containsMatchingElement(
+        <span>Disabled</span>
+      )).toBeTruthy();
+      expect(disabled.props().disabled).toEqual(true);
+    });
+  });
+
+  describe('Passing a custom onClick', () => {
+    it('triggers when the button is clicked', () => {
+      const wrapper = render({
+        children: 'foo',
+        onClick: jest.fn()
+      });
+
+      wrapper.simulate('click');
+      expect(wrapper.props().onClick).toHaveBeenCalled();
```

Comment:
this test is redundant - when triggering a click on the wrapper element React will always call the `onClick` method, you should trigger that event on a descendant of this component.